rails/app/models/external_activity.rb

```diff
@@ -299,20 +295,13 @@ def run_format
     :run_resource_html
   end
 
-  def has_launch_url?
-    !launch_url.blank?
-  end
-
-  def lara_activity?
-    material_type == 'Activity' && has_launch_url?
-  end
-
-  def lara_sequence?
-    material_type == 'Investigation' && has_launch_url?
-  end
-
   def lara_activity_or_sequence?
-    lara_activity? || lara_sequence?
+    tool = Tool.where(id: self.tool_id).first
+    if tool && tool.source_type == 'LARA'
```

Comment:
```suggestion
    if self.tool && self.tool.source_type == 'LARA'
```

The external activity models defines a `belongs_to :tool` association. So you can just call self.tool (or if you want to be more concise you can drop the `self.`)